packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,424 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import cx from 'classnames';
+import {
+  InputFlexContainer,
+  useTheme,
+  SvgSearch,
+  SvgCloseSmall,
+  useSafeContext,
+  useId,
+  Icon,
+  useMergedRefs,
+} from '../utils/index.js';
+import type {
+  IconProps,
+  PolymorphicForwardRefComponent,
+  PolymorphicComponentProps,
+  InputFlexContainerProps,
+} from '../utils/index.js';
+import { IconButton } from '../Buttons/IconButton/index.js';
+import type { IconButtonProps } from '../Buttons/IconButton/index.js';
+import '@itwin/itwinui-css/css/searchbox.css';
+
+const SearchBoxContext = React.createContext<
+  | {
+      /**
+       * Context prop for sizing subcomponents
+       */
+      size?: 'small' | 'large';
+      /**
+       * Context prop for disabling subcomponents
+       */
+      isDisabled?: boolean;
+      /**
+       * Id to pass to input
+       */
+      inputId: string;
+      setInputId: (inputId: string) => void;
```

Comment:
`setInputId` is missing a comment

Reply:
It is for context so internal use only.
Added comment.